datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java

```diff
@@ -165,9 +180,11 @@ public void initIndexColumnConverters(CarbonTable carbonTable, List<CarbonColumn
       for (CarbonBloomFilter bloomFilter : bloomIndexList) {
         boolean scanRequired = bloomFilter.membershipTest(new Key(bloomQueryModel.filterValue));
         if (scanRequired) {
+          String blockletNo =
+              isBlockletCacheLevel ? String.valueOf(bloomFilter.getBlockletNo()) : "-1";
```

Comment:
I think Bloom dataMap should return the actual blocklet Id. The old behavior should not be modified based on cache level

Reply:
ok, fixed